tests/syntax/test_parser.py

```diff
@@ -336,3 +336,849 @@ def test_specifier_apparently_facing_from(self):
+                    ),
+                    LShift(),
+                    Name("D", Load()),
+                ),
+            ),
+            (
+                "angle from A + B to C + D",
+                AngleFromOp(
+                    BinOp(Name("C", Load()), Add(), Name("D", Load())),
+                    BinOp(Name("A", Load()), Add(), Name("B", Load())),
+                ),
+            ),
+            (
+                "angle to A + B",
+                AngleFromOp(
+                    BinOp(Name("A", Load()), Add(), Name("B", Load())),
+                    None,
+                ),
+            ),
+            (
+                "angle from A + B",
+                AngleFromOp(
+                    None,
+                    BinOp(Name("A", Load()), Add(), Name("B", Load())),
+                ),
+            ),
+            (
+                "angle to A << B",
+                BinOp(
+                    AngleFromOp(Name("A", Load()), None),
+                    LShift(),
+                    Name("B", Load()),
+                ),
+            ),
+            (
+                "angle from A << B",
+                BinOp(
+                    AngleFromOp(None, Name("A", Load())),
+                    LShift(),
+                    Name("B", Load()),
+                ),
+            ),
+        ],
+    )
+    def test_angle_from_precedence(self, code, expected):
+        mod = parse_string_helper(code)
+        stmt = mod.body[0].value
+        assert dump(stmt, annotate_fields=False) == dump(
+            expected, annotate_fields=False
+        )
+
+    def test_follow(self):
+        mod = parse_string_helper("follow x from y for z")
+        stmt = mod.body[0]
+        match stmt:
+            case Expr(FollowOp(Name("x"), Name("y"), Name("z"))):
+                assert True
+            case _:
+                assert False
+
+    @pytest.mark.parametrize(
+        "code,expected",
+        [
+            (
+                "follow A + B from C + D for E + F",
+                FollowOp(
+                    BinOp(Name("A", Load()), Add(), Name("B", Load())),
+                    BinOp(Name("C", Load()), Add(), Name("D", Load())),
+                    BinOp(Name("E", Load()), Add(), Name("F", Load())),
+                ),
+            ),
+            (
+                "follow A << B from C << D for E << F",
+                BinOp(
+                    FollowOp(
+                        BinOp(Name("A", Load()), LShift(), Name("B", Load())),
+                        BinOp(Name("C", Load()), LShift(), Name("D", Load())),
+                        Name("E", Load()),
+                    ),
+                    LShift(),
+                    Name("F", Load()),
+                ),
+            ),
+        ],
+    )
+    def test_follow_precedence(self, code, expected):
+        mod = parse_string_helper(code)
+        stmt = mod.body[0].value
+        assert dump(stmt, annotate_fields=False) == dump(
```

Comment:
This approach of using `dump` to compare the ASTs is used frequently in this file, and I'm a little worried it might not be stable enough for this purpose, since `dump` is intended for debugging. Can we define a helper function `compareASTs` or something so that we can adjust all of these tests uniformly in one place if needed later? It would also simplify the code.